### PR TITLE
Metrics Test Pipeline Integration

### PR DESCRIPTION
**Pull request by and-rewsmith**
This PR adds metrics collector to the test pipeline.

### CODE REVIEW

File: e2e_deployment_files/long_haul_deployment.template.windows.json [hunk 1]
```diff
@@ -23,17 +23,31 @@
...
+                  "value": "true"
+              }
             }
           },
           "edgeHub": {
             "type": "docker",
             "settings": {
               "image": "<Container_Registry>/microsoft/azureiotedge-hub:<Build.BuildNumber>-windows-<Architecture>",
-              "createOptions": "{\"HostConfig\": {\"PortBindings\": {\"8883/tcp\": [{\"HostPort\": \"8883\"}],\"443/tcp\": [{\"HostPort\": \"443\"}],\"5671/tcp\": [{\"HostPort\": \"5671\"}]}}}"
+              "createOptions": "{\"User\": \"ContainerAdministrator\",\"HostConfig\": {\"PortBindings\": {\"8883/tcp\": [{\"HostPort\": \"8883\"}],\"443/tcp\": [{\"HostPort\": \"443\"}],\"5671/tcp\": [{\"HostPort\": \"5671\"}]}}}"
```
Comment (and-rewsmith) [inline e2e_deployment_files/long_haul_deployment.template.windows.json]: On windows, we need this for metrics to work.

File: builds/e2e/templates/longhaul-deploy-windows.yaml [hunk 2]
```diff
@@ -68,5 +71,8 @@ steps:
           -LogAnalyticsSharedKey "${{ parameters['logAnalytics.sharedKey'] }}" `
           -TwinUpdateSize "${{ parameters['twinTester.twinUpdateSize'] }}" `
           -TwinUpdateFrequency "${{ parameters['twinTester.twinUpdateFrequency'] }}" `
-          -TwinUpdateFailureThreshold "${{ parameters['twinTester.twinUpdateFailureThreshold'] }}"
+          -TwinUpdateFailureThreshold "${{ parameters['twinTester.twinUpdateFailureThreshold'] }}" `
+          -MetricsEndpointsCSV "${{ parameters['metricsCollector.metricsEndpointsCSV'] }}" `
+          -MetricsScrapeFrequencyInSecs "${{ parameters['metricsCollector.scrapeFrequencyInSecs'] }}" `
+          -MetricsUploadTarget "${{ parameters['metricsCollector.uploadTarget'] }}"
```
Comment (philipktlin) [inline builds/e2e/templates/longhaul-deploy-windows.yaml]: seems -cleanAll option is missing here.  Please help to fix it.

File: builds/e2e/templates/stresstest-deploy-windows.yaml [hunk 3]
```diff
@@ -76,7 +79,10 @@ steps:
           -LogAnalyticsSharedKey "${{ parameters['logAnalytics.sharedKey'] }}" `
           -TwinUpdateSize "${{ parameters['twinTester.twinUpdateSize'] }}" `
           -TwinUpdateFrequency "${{ parameters['twinTester.twinUpdateFrequency'] }}" `
-          -TwinUpdateFailureThreshold "${{ parameters['twinTester.twinUpdateFailureThreshold'] }}"
+          -TwinUpdateFailureThreshold "${{ parameters['twinTester.twinUpdateFailureThreshold'] }}" `
+          -MetricsEndpointsCSV "${{ parameters['metricsCollector.metricsEndpointsCSV'] }}" `
+          -MetricsScrapeFrequencyInSecs "${{ parameters['metricsCollector.scrapeFrequencyInSecs'] }}" `
+          -MetricsUploadTarget "${{ parameters['metricsCollector.uploadTarget'] }}"
```
Comment (philipktlin) [inline builds/e2e/templates/stresstest-deploy-windows.yaml]: Same here; please help to add -cleanAll option

File: e2e_deployment_files/long_haul_deployment.template.windows.json (hunk 1 above)
Comment (philipktlin) [reply e2e_deployment_files/long_haul_deployment.template.windows.json]: is it possible to add comment here?

Comment (and-rewsmith) [reply e2e_deployment_files/long_haul_deployment.template.windows.json]: According to [this post](https://stackoverflow.com/questions/244777/can-comments-be-used-in-json)

**Comment by philipktlin** [comment]
Please test it before commit

**Pull request merged**